src/hardware/memory.cpp

```diff
@@ -1825,7 +1825,7 @@ void Init_RAM() {
 
     /* Allocate the RAM. We alloc as a large unsigned char array. new[] does not initialize the array,
      * so we then must zero the buffer. */
-    MemBase = new uint8_t[memory.pages*4096];
+    MemBase = new(std::nothrow) uint8_t[memory.pages*4096];
```

Comment:
Good catch!

(std::nothrow) makes sure that if the new operator cannot allocate memory, it returns NULL instead of throwing a std::bad_alloc exception.